lib/runner/variant-reassignment.js

```diff
@@ -277,20 +303,41 @@ export default class VariantReassignment {
   }
 
   /**
-   * Select products that has at least one variant from the productDraft.
+   * Select products that has at least one variant
+   * or at least one matching slug with the productDraft.
    * @param productDraft
-   * @param products
+   * @param products if present, then this will be used and no fetching from CTP is needed
    * @returns {*}
    * @private
    */
   _selectMatchingProducts (productDraft, products) {
     const productDraftSkus = this.productService.getProductDraftSkus(productDraft)
+    const productDraftSlugs = productDraft.slug
     if (products) {
-      const skuToProductMap = this._createSkuToProductMap(products)
-      const matchingProducts = productDraftSkus.map(sku => skuToProductMap.get(sku))
-      return _.uniq(matchingProducts)
+      // select products that matches by at least one slug with productDraft
+      const matchingProductsBySlug = this._selectMatchingProductsBySlug(products, productDraft)
+      // select products that matches by at least one variant with productDraft
+      const matchingProductsBySku = this._selectMatchingProductsByVariant(products,
+        productDraftSkus)
+      return _.compact(_.uniq(matchingProductsBySku.concat(matchingProductsBySlug)))
     }
-    return this.productService.getProductsBySkus(productDraftSkus)
+    return this.productService.getProductsBySkusOrSlugs(productDraftSkus, productDraftSlugs)
+  }
+
+  _selectMatchingProductsBySlug (products, productDraft) {
+    return products.filter((product) => {
+      const productSlug = product.masterData.staged.slug
```

Comment:
When loading matching products we also fetch products by `current.slug` but here we compare only staged slugs, is it correct?

Reply:
https://github.com/commercetools/commercetools-node-variant-reassignment/pull/45/commits/c2f06e525c772fb87ed8e2267206efab677ccb70#diff-eda63119a30637151d12307cefd81cf0R427